engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -465,6 +471,49 @@ private Class<?> findNestedClass(Class<?> enclosingClass, String nestedClassName
         return m.get(nestedClassName);
     }
 
+    private Method getMethod(final Class<?> scope, final MethodCallExpr methodCallExpr, final Class<?>[] paramTypes,
+            final Class<?>[][] parameterizedTypes) {
+        final String methodName = methodCallExpr.getNameAsString();
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+
+        if (scope == null) {
+            final Class<?> methodClass = variables.get(methodName);
+            if (methodClass == PyCallableWrapper.class) {
+                final Object paramValueRaw = queryScope.readParamValue(methodName, null);
+                final PyCallableWrapper pyCallableWrapper = (PyCallableWrapper) paramValueRaw;
+                try {
+                    String javaMethodName = pyCallableWrapper.getAttribute("_j_simple_name", String.class);
+                    methodCallExpr.setName(javaMethodName);
+                    return getMethod(null, javaMethodName, paramTypes, parameterizedTypes);
+                } catch (RuntimeException e) {
+                    // Not, swappable function, can safely ignore
+                }
+            }
+        } else if (scope == org.jpy.PyObject.class || scope == PyCallableWrapper.class) {
+            String pythonExpr = methodCallExpr.getScope().get().toString() + "." + methodName;
+            PythonDeephavenSession pds = (PythonDeephavenSession) ((AbstractScriptSession.UnsynchronizedScriptSessionQueryScope) queryScope).scriptSession();
```

Comment:
This is gross - and at least going by other usages of this, we might need to hold a lock for it.